### PR TITLE
Add CVE-2022-0001 and CVE-2022-0002 (Linux only)

### PR DESCRIPTION
**Pull request by antoniogi**
Check the status of the mitigations indicated by the 'Branch History Injection and Intra-mode Branch Target Injection: https://www.intel.com/content/www/us/en/developer/articles/technical/software-security-guidance/technical-documentation/branch-history-injection.html

### CODE REVIEW

**Comment by speed47** [comment]
Hello, and thanks for your PR!
I'll try to find some time in the next few days to review it, as I'd like support for these new vulns be available ASAP in the script.

**Comment by antoniogi** [comment]
No worries. I'm thinking the basic functionality is there, but we might need to add more code for extra info, like 'please update your OS'.

**Comment by antoniogi** [comment]
@speed47 do you need any help with this PR?

File: spectre-meltdown-checker.sh [hunk 1]
```diff
@@ -82,7 +82,7 @@ show_usage()
 		--batch short		produce only one line with the vulnerabilities separated by spaces
 		--batch json		produce JSON output formatted for Puppet, Ansible, Chef...
 		--batch nrpe		produce machine readable output formatted for NRPE
-		--batch prometheus      produce output for consumption by prometheus-node-exporter
+		--batch prometheus	produce output for consumption by prometheus-node-exporter
```
Comment (aliceinwire) [inline spectre-meltdown-checker.sh:85]: can you fix the code format style?

File: spectre-meltdown-checker.sh [hunk 2]
```diff
@@ -288,7 +288,9 @@ cve2name()
 		CVE-2019-11091) echo "RIDL, microarchitectural data sampling uncacheable memory (MDSUM)";;
 		CVE-2019-11135) echo "ZombieLoad V2, TSX Asynchronous Abort (TAA)";;
 		CVE-2018-12207) echo "No eXcuses, iTLB Multihit, machine check exception on page size changes (MCEPSC)";;
-		CVE-2020-0543) echo "Special Register Buffer Data Sampling (SRBDS)";;
+		CVE-2020-0543)  echo "Special Register Buffer Data Sampling (SRBDS)";;
```
Comment (aliceinwire) [inline spectre-meltdown-checker.sh:295]: Also, here looks like you are mixing formatting fix with CVE-2022-0001 and CVE-2022-0002 checking commit

File: spectre-meltdown-checker.sh [hunk 3]
```diff
@@ -1112,11 +1137,11 @@ while [ -n "$1" ]; do
 			exit 255
 		fi
 		case "$2" in
-			1)		opt_cve_list="$opt_cve_list CVE-2017-5753"; opt_cve_all=0;;
-			2)		opt_cve_list="$opt_cve_list CVE-2017-5715"; opt_cve_all=0;;
-			3)		opt_cve_list="$opt_cve_list CVE-2017-5754"; opt_cve_all=0;;
-			3a)		opt_cve_list="$opt_cve_list CVE-2018-3640"; opt_cve_all=0;;
-			4)		opt_cve_list="$opt_cve_list CVE-2018-3639"; opt_cve_all=0;;
+			1)	opt_cve_list="$opt_cve_list CVE-2017-5753"; opt_cve_all=0;;
```
Comment (aliceinwire) [inline spectre-meltdown-checker.sh:1158]: Also here  CVE-2022-0001 and CVE-2022-0002 checking and format fix are mixed.

File: spectre-meltdown-checker.sh [hunk 4]
```diff
@@ -3916,7 +4060,6 @@ check_CVE_2017_5715_linux()
 
 		# if we arrive here and didn't already call pvulnstatus, then it's VULN, let's explain why
 		if [ "$pvulnstatus_last_cve" != "$cve" ]; then
-			# explain what's needed for this CPU
```
Comment (aliceinwire) [inline spectre-meltdown-checker.sh:4096]: ?

Comment (aliceinwire) [reply spectre-meltdown-checker.sh:4096]: Probably also this should go in a different commit

**Comment by aliceinwire** [comment]
I think this link https://cdrdv2.intel.com/v1/dl/getContent/691322 need some sort of account registration

File: spectre-meltdown-checker.sh (hunk 1 above)
Comment (antoniogi) [reply spectre-meltdown-checker.sh]: Should be done now.

File: spectre-meltdown-checker.sh (hunk 2 above)
Comment (antoniogi) [reply spectre-meltdown-checker.sh]: Uggh. You are right. Thanks.

File: spectre-meltdown-checker.sh (hunk 4 above)
Comment (antoniogi) [reply spectre-meltdown-checker.sh]: That was an ooops. Great catch.It shouldn't have been part of this (was an original prototype).

**Comment by aliceinwire** [comment]
Just squash the changes in the previous BHI commit, instead of adding a new reversing commit. And create a new commit for the cosmetic ones.

**Pull request closed without merging**